pkgs/by-name/li/libation/package.nix

```
@@ -0,0 +1,96 @@
+{ lib
+, stdenv
+, buildDotnetModule
+, fetchFromGitHub
+, dotnetCorePackages
+, wrapGAppsHook
+
+, libX11
+, libICE
+, libSM
+, libXi
+, libXcursor
+, libXext
+, libXrandr
+, fontconfig
+, glew
+, gtk3
+}:
+
+buildDotnetModule rec {
+  pname = "libation";
+  version = "11.1.0";
+
+  src = fetchFromGitHub {
+    owner = "rmcrackan";
+    repo = "Libation";
+    rev = "v${version}";
+    hash = "sha256-NxG1H8lj+aBpgKj03CDpX/tLT0SxDS3pnZGQ2ultBnQ=";
+  };
+
+  sourceRoot = "${src.name}/Source";
+
+  dotnet-sdk = dotnetCorePackages.sdk_7_0;
+  dotnet-runtime = dotnetCorePackages.runtime_7_0;
+
+  nugetDeps = ./deps.nix;
+
+  dotnetFlags = [ "-p:PublishReadyToRun=false" ];
+
+  projectFile = [
+    "LibationAvalonia/LibationAvalonia.csproj"
+    "LibationCli/LibationCli.csproj"
+    "HangoverAvalonia/HangoverAvalonia.csproj"
+  ];
+
+  nativeBuildInputs = [ wrapGAppsHook ];
+
+  runtimeDeps = [
+    # For Avalonia UI
+    libX11
+    libICE
+    libSM
+    libXi
+    libXcursor
+    libXext
+    libXrandr
+    fontconfig
+    glew
+    # For file dialogs
+    gtk3
+  ];
+
+  postInstall = ''
+    install -Dm644 LoadByOS/LinuxConfigApp/libation_glass.svg $out/share/icons/hicolor/scalable/apps/${pname}.svg
+    install -Dm644 LoadByOS/LinuxConfigApp/Libation.desktop $out/share/applications/${pname}.desktop
+    substituteInPlace $out/share/applications/${pname}.desktop \
+        --replace "/usr/bin/libation" "${meta.mainProgram}"
+  '';
+
+  # wrap manually, because we need lower case excutables
+  dontDotnetFixup = true;
+
+  preFixup = ''
+    # remove binaries for other platform, like upstream does
+    pushd $out/lib/${pname}
+    rm -f *.x86.dll *.x64.dll
```

```suggestion
    rm -f *.dll
```
Still a bunch of files left

Also, is the `-f` needed?

---

I don't know what I can do about those files...
Could you send me the remaining files, that should be temoved (that are not dotnet .dll files (those are different from normal windows dlls and are needed for all platforms))

About the -f, it looked like one of the possible library endings didn't actually appear (maybe arm64.so, i dont remember exactly)
But that may not be the case when building on other systems and I only have access to x64 linux. That's why -f is there: dont error if it doesnt exist

---

I have 3 normal `.so` files which do not have a prefix:
`libe_sqlite3.so` `libHarfBuzzSharp.so` and `libSkiaSharp.so`
would be great to know if they are `.dylib` files on darwin or not...
Other than this I think the current logic covers everything

---

> that are not dotnet .dll files (those are different from normal windows dlls and are needed for all platforms)

Ah, `file` tells me those are `PE32 executable (DLL)`, so I figured they were actual DLLs. Dotnet is weird.

I do not have darwin, so i cannot test that.